src/Feature/Editor/Draw.re

```diff
@@ -94,13 +89,19 @@ let drawShapedText = {
 
   Skia.Paint.setLcdRenderText(paint, true);
 
-  (~context, ~x, ~y, ~color, text) => {
-    let text =
-      Revery.Font.(shape(context.font, text) |> ShapeResult.getGlyphString);
+  (~context, ~x, ~y, ~color, ~bold, ~italic, text) => {
+    let font =
+      Revery.Font.Family.resolve(
+        ~italic,
+        bold ? Revery.Font.Weight.Bold : Revery.Font.Weight.Normal,
+        context.fontFamily,
+      )
+      |> Stdlib.Result.get_ok;
```

Comment:
We need to be careful with the `Stdlib.Result.get_ok` method - in the case where, for some reason, the `resolve returns an `Error` - this would manifest as a crash.

Seems like this could happen a few ways:
1) The user switches a theme, the application crashes because it couldn't resolve a now-bold or now-italic token
2) The user switches a fonts, and the application crashes
3) ...even worse - the user types a token, it gets resolved as bold / italic, causing a crash and the user losing work.

3 would be the worst case! We should avoid `Stdlib.Result.get_ok` during runtime, and handle the `result`. I can see two possibilities:
- Possibility 1: This doesn't actually need to be `result` - it always returns _something_ (ie, a normal font), even if couldn't resolve the bold/italic version. Or, we could provide an alternative method that is like `resolveOrFallback` - that tries to return `bold/italic`, but gives the normal variant if it can't find it. That wouldn't need to return a `result`.
- Possibility 2 - there isn't a good a way to handle when there isn't an italic / bold variant, and in that case, the best thing we could do is just avoid rendering (ie, use `Result.iter` to do the rendering).

I think we should be able to have a non-result API that falls back to the 'normal' variant if it can't resolve the `bold`/`italic`/`both` variant, but either way we shouldn't have a chance of crashing here.

Reply:
That sounds great to me! Would that API be housed in `Service_Font`, as you see it?

Reply:
> Would that API be housed in Service_Font, as you see it?

Good question. I was actually thinking it could make sense for [`FontFamily`](https://github.com/revery-ui/revery/blob/master/src/Font/FontFamily.rei) - a `resolve` that returns a result (a possibility of failure), and a `resolveWithFallback` which always returns a `FontCache.t` - just, in the case where bold/italic/variant isn't available, we'd use the default cache.

I'd be fine putting it in `Service_Font` for now, too, as an intermediate to try out - and when we feel good about the API (or see another use case in Revery), we could push it back at that time.

Reply:
Awesome, for the time being I put it in `ServiceFont` as `resolveWithFallback` that resolves to the default font in the event of an error. Let me know if you think this is satisfactory!